### PR TITLE
refactor: post options menu removing bookmark

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Bookmark on card should be based on the experiment not on the feed layout.
- For the post options menu, it should be the same.

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-20 #done

### CODE REVIEW

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 1]
```diff
@@ -285,7 +287,12 @@ export default function PostOptionsMenu({
       label: 'Hide',
       action: onHidePost,
     },
-    {
+  ];
+
+  const bookmarkOnCard = useFeature(feature.bookmarkOnCard);
```
Comment (rebelchris) [inline packages/shared/src/components/PostOptionsMenu.tsx:292]: We have this currently in the post option menu dynamically?
Cant recall if we just always show it or based on experiment

Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:292]: From the ticket Nensi created, she [mentioned](https://dailydotdev.slack.com/archives/C061T644XV3/p1700551627401439) it to be dynamically available.

Comment (nensidosari) [reply packages/shared/src/components/PostOptionsMenu.tsx:293]: Yup, this should be dynamically available also in the share post options menu

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 2]
```diff
@@ -295,8 +303,11 @@ export default function PostOptionsMenu({
       ),
       label: `${post?.bookmarked ? 'Remove from' : 'Save to'} bookmarks`,
       action: onToggleBookmark,
-    },
-    {
+    });
+  }
+
+  if (shouldUseFeedLayoutV1) {
```
Comment (nensidosari) [inline packages/shared/src/components/PostOptionsMenu.tsx:309]: shouldn't this be `!shouldUseFeedLayoutV1` ?

File: packages/shared/src/components/PostOptionsMenu.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:293]: Cool. Let me apply it on that end too.

File: packages/shared/src/components/PostOptionsMenu.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:309]: Oh, right. Will update quickly.

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 3]
```diff
@@ -109,24 +100,6 @@ export default function PostOptionsMenu({
     shouldInvalidateQueries: false,
   });
 
-  const { toggleBookmark } = useBookmarkPost({
```
Comment (nensidosari) [inline packages/shared/src/components/PostOptionsMenu.tsx:112]: Can we make this at the moment removed if in feed layout experiment just like we're doing for downvote?

File: packages/shared/src/components/PostOptionsMenu.tsx [hunk 4]
```diff
@@ -285,18 +258,12 @@ export default function PostOptionsMenu({
       label: 'Hide',
       action: onHidePost,
     },
-    {
-      icon: (
```
Comment (nensidosari) [inline packages/shared/src/components/PostOptionsMenu.tsx:289]: Same as above here

File: packages/shared/src/components/ShareOptionsMenu.tsx [hunk 5]
```diff
@@ -59,17 +56,6 @@ export default function ShareOptionsMenu({
       text: 'Share post via...',
       action: () => onShare(post),
     },
-    {
```
Comment (nensidosari) [inline packages/shared/src/components/ShareOptionsMenu.tsx:62]: Same as above

**Review by nensidosari — COMMENTED** [review]
Looks great, just one small ask 🙏 We want these removals to be behind the feed layout experiment, as at the moment we don't know what the decision is going to be from other experiments.

File: packages/shared/src/components/PostOptionsMenu.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:112]: The bookmark is going to be regardless of feed layout. Either control or v1 will have the bookmark on card once it wins.

Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:112]: I thought we were under the assumption that the bookmark on card wins? That was already what we had before.

Comment (sshanzel) [reply packages/shared/src/components/PostOptionsMenu.tsx:112]: Ah, right. We only do that assumption on v1.

**Review by nensidosari — APPROVED** [review]
Looks great, we'll check those conditions once the bookmark experiment is concluded. 🥳 
Apologies for the confusion, having multiple experiments introducing the same thing is confusing.

**Pull request merged**